dbms/src/Flash/tests/gtest_execution_summary.cpp

```diff
@@ -0,0 +1,139 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Coprocessor/ExecutionSummaryCollector.h>
+#include <TestUtils/ExecutorTestUtils.h>
+#include <TestUtils/mockExecutor.h>
+
+namespace DB
+{
+namespace tests
+{
+class ExecutionSummaryTestRunner : public DB::tests::ExecutorTest
+{
+public:
+    void initializeContext() override
+    {
+        ExecutorTest::initializeContext();
+        context.addMockTable({"test_db", "test_table"},
+                             {{"s1", TiDB::TP::TypeString}, {"s2", TiDB::TP::TypeString}},
+                             {toNullableVec<String>("s1", {"banana", {}, "banana", "banana", {}, "banana", "banana", {}, "banana", "banana", {}, "banana"}),
+                              toNullableVec<String>("s2", {"apple", {}, "banana", "apple", {}, "banana", "apple", {}, "banana", "apple", {}, "banana"})});
+        context.addExchangeReceiver("test_exchange",
+                                    {{"s1", TiDB::TP::TypeString}, {"s2", TiDB::TP::TypeString}},
+                                    {toNullableVec<String>("s1", {"banana", {}, "banana", "banana", {}, "banana", "banana", {}, "banana", "banana", {}, "banana"}),
+                                     toNullableVec<String>("s2", {"apple", {}, "banana", "apple", {}, "banana", "apple", {}, "banana", "apple", {}, "banana"})});
+    }
+
+    static constexpr size_t concurrency = 10;
+    static constexpr int not_check_rows = -1;
+    // <rows, concurrency>
+    using ProfileInfo = std::pair<int, size_t>;
+    using Expect = std::unordered_map<String, ProfileInfo>;
+    void testForExecutionSummary(
+        const std::shared_ptr<tipb::DAGRequest> & request,
+        const Expect & expect)
+    {
+        request->set_collect_execution_summaries(true);
+        DAGContext dag_context(*request, "test_execution_summary", concurrency);
+        executeStreams(&dag_context);
+        ASSERT_EQ(dag_context.getProfileStreamsMap().size(), expect.size());
+        ASSERT_TRUE(dag_context.collect_execution_summaries);
+        ExecutionSummaryCollector summary_collector(dag_context);
+        auto summaries = summary_collector.genExecutionSummaryResponse().execution_summaries();
+        ASSERT_EQ(summaries.size(), expect.size());
+        for (const auto & summary : summaries)
+        {
+            ASSERT_TRUE(summary.has_executor_id());
+            auto it = expect.find(summary.executor_id());
+            ASSERT_TRUE(it != expect.end()) << fmt::format("unknown executor_id: {}", summary.executor_id());
+            if (it->second.first != not_check_rows)
+                ASSERT_EQ(summary.num_produced_rows(), it->second.first) << fmt::format("executor_id: {}", summary.executor_id());
```

Comment:
How to check num_interations?

Reply:
Because num_iterations depends on the size of the block and the amount of data... So it is not checked here...
But it can be discussed...